app/invocation/invocation_action_card.tsx

```diff
@@ -28,43 +37,48 @@ export default class InvocationActionCardComponent extends React.Component<Props
     let actionFile = "bytestream://" + this.getCacheAddress() + "/blobs/" + this.props.search.get("actionDigest");
     rpcService
       .fetchBytestreamFile(actionFile, this.props.model.getId(), "arraybuffer")
-      .then((action_buff: any) => {
-        let tempAction = build.bazel.remote.execution.v2.Action.decode(new Uint8Array(action_buff));
+      .then((buffer: any) => {
         this.setState({
-          ...this.state,
-          contents: action_buff,
-          action: tempAction,
+          action: build.bazel.remote.execution.v2.Action.decode(new Uint8Array(buffer)),
         });
-        this.fetchCommand(tempAction);
+        this.fetchCommand(this.state.action);
```

Comment:
React doesn't always guarantee that `setState` will immediately update the state -- https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

> Think of setState() as a request rather than an immediate command to update the component. For better perceived performance, React may delay it, and then update several components in a single pass. React does not guarantee that the state changes are applied immediately.

So you could put `action` in a local variable then use that local instead of `this.state.action`